### ISSUE TITLE
[RF] RooLinearVar not used in plot projection integrals

### ISSUE THREAD

**Issue by elusian** [root-project/root#12646]
- [x] Checked for duplicates

### Describe the bug

`RooLinearVar`s and other `RooAbsRealLValue`s are special inside integrals, as `RooRealIntegral` over the underling `RooRealVar` can transparently integrate over the `RooLinearVar` variable and then use the jacobian to correct the result.
If the integral is inside a plot instead, the integral does not see the `RooLinearVar` as client of the integration variable, and decides to just integrate numerically, which, at least in my case, is slow.

### Expected behavior

`RooLinearVar` are used in the integration even when doing a plot projection.

### To Reproduce

```python
import ROOT
from ROOT.RooFit import RooConst

x = ROOT.RooRealVar('x', '', 0, 1)
y = ROOT.RooRealVar('y', '', 0, 1)

x2 = ROOT.RooLinearVar('x2', '', x, RooConst(1), RooConst(0))
y2 = ROOT.RooLinearVar('y2', '', y, RooConst(1), RooConst(0))

# RooGaussian can integrate over x or mu, but not both
# still, the issue is visible regardless
D2 = ROOT.RooGaussian('g', '', x2, y2, RooConst(0.2))

# needed only to set the correct normalization in the frame
dt = D2.generate({x,y}, NumEvents = 10)

# works fine, numerical integral only for the 2D normalization integral, and its over y2 (meaning that the RooLinearVar integral works)
# but not in the y projection
# if the pdf supported 2D integrals, there would be no numerical integrals at all
integral = D2.createIntegral({y}, NormSet = {x,y})
print('Projection is', integral, flush = True)

# numerical integral over y2 in the 2D normalization (expected, like above)
# numerical integral in the y projection (unwanted)
frame = x.frame()
dt.plotOn(frame)
D2.plotOn(frame)
```

### Setup

ROOT master from LCG dev3

**Comment by guitargeek** [comment]
Thanks for reporting this! I'm on it. It's another classic RooFit bug. RooFit should consider all RooAbsArgs with the same name as equivalent, but sometimes it doesn't. Here, cloning the observables, which is done in the plotting, messes up something in the integral creation:

```C++
#include <RooConstVar.h>
#include <RooDataSet.h>
#include <RooGaussian.h>
#include <RooLinearVar.h>
#include <RooRealVar.h>

void repro()
{
   using namespace RooFit;

   RooRealVar x{"x", "", 0, 1};
   RooRealVar y{"y", "", 0, 1};

   RooLinearVar x2{"x2", "", x, RooConst(1), RooConst(0)};
   RooLinearVar y2{"y2", "", y, RooConst(1), RooConst(0)};

   // RooGaussian can integrate over x or mu, but not both still, the issue is
   // visible regardless
   RooGaussian gauss{"g", "", x2, y2, RooConst(0.2)};

   // works fine, numerical integral only for the 2D normalization integral,
   // and its over y2 (meaning that the RooLinearVar integral works) but not in
   // the y projection if the pdf supported 2D integrals, there would be no
   // numerical integrals at all
   std::unique_ptr<RooAbsReal> integral{gauss.createIntegral({y}, {x, y})};
   integral->Print();

   RooRealVar xCopy{x};
   RooRealVar yCopy{y};

   // numerical integral over y (unwanted)
   std::unique_ptr<RooAbsReal> integral2{gauss.createIntegral({yCopy}, {xCopy, yCopy})};
   integral2->Print();
}
```

I suppose the fix will be a small change in RooRealIntegral to compare something by name and not by pointer.

**Comment by elusian** [comment]
It's a bit more involved than that, from what I could understand from debugging: `RooRealIntegral` searches for the clients of the integration variables, and the plot variables have none (clone of the dataset variables?). 
Overriding the projection variables by passing `Project` to the pdf plot solves this, and then it becomes an actual pointer comparison problem as you say.

**Comment by guitargeek** [comment]
Yes, a little bit more involved but not much :) I opened a PR, it also fixes your initial reproducer. I'll also backport it to the 6.28 branch so it gets included in 6.28.04
